src/cdk/table/table.ts

```diff
@@ -497,11 +519,21 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
       @Optional() @Inject(_COALESCED_STYLE_SCHEDULER)
         protected readonly _coalescedStyleScheduler?: _CoalescedStyleScheduler,
       @Optional() @SkipSelf() @Inject(STICKY_POSITIONING_LISTENER)
-      protected readonly _stickyPositioningListener?: StickyPositioningListener,
+      protected readonly _parentPositioningListener?: StickyPositioningListener,
       // Optional for backwards compatibility. The viewport ruler is provided in root. Therefore,
       // this property will never be null.
       // tslint:disable-next-line: lightweight-tokens
-      @Optional() private readonly _viewportRuler?: ViewportRuler) {
+      @Optional() private readonly _viewportRuler?: ViewportRuler,
+      @Optional() @Inject(STICKY_POSITIONING_LISTENER)
+      protected readonly _positioningListener?: StickyPositioningListener) {
+    // FIXME DO NOT SUBMIT: Is the following behavior acceptable?
+    //
+    // The table will override the StickyPositioningListener provider to `null` to prevent child
+    // tables from inheriting it. However, if a directive on the table configures a new provider, it
+    // should be used instead. Therefore, when a directive on the table configures a positioning
+    // listener provider, the provider will be inherited by child tables.
+    this._stickyPositioningListener = this._positioningListener ?? this._parentPositioningListener;
```

Comment:
To clarify that I understand this correctly: the goal here is to support multiple different directives adding their own sticky position listener? If that's the case, we should probably refactor this in one of two ways:
* Have one "notifier" that all those directives subscribe to
* Change this to a multi provider and let multiple directives register listeners

I would definitely lean towards the first option here as being more in line with the rest of the codebase

Reply:
The table currently retrieves the `StickyPositioningListener` provider from the parent component and overrides the provider to `null` to prevent child tables from inheriting it.

```
<parent-with-listener>
  <table></table>
</parent-with-listener>
```

However, the virtual scroll directive is bound directly on the table, so the table must get the provider from `@Self()` when present.

```
<table virtualScroll></table>
```